### PR TITLE
feat(cloudfront): Match console rec for ACM cert Security Policy (TLSv1.1_2016)

### LINKED ISSUE #6547: Cloudfront Web Distribution Security Policy default does not match console recommendation
Currently, when using SNI, CDK defaults to TLSv1:

https://github.com/aws/aws-cdk/blob/0f0e2055cab08551bc6e5dfb8a1b6219368263c5/packages/%40aws-cdk/aws-cloudfront/lib/web_distribution.ts#L64-L71

However, the console recommends using TLSv1.1_2016:

Screen Shot 2020-03-02 at 1 36 20 PM

> Choose the security policy that you want CloudFront to use for HTTPS connections. A security policy determines two settings: the SSL/TLS protocol that CloudFront uses to communicate with viewers, and the cipher that CloudFront uses to encrypt the content that it returns to viewers. When you use a custom SSL certificate and SNI, you must use TLSv1 or later. We recommend that you specify TLSv1.1_2016 unless your users are using browsers or devices that do not support TLSv1.1 or later. When you use a custom SSL certificate and dedicated IP addresses, we recommend that you use TLSv1. In this configuration, the TLSv1_2016, TLSv1.1_2016 and TLSv1.2_2018 security policies are not available.

### Reproduction Steps

```ts
import { CloudFrontWebDistribution } from '@aws-cdk/aws-cloudfront'

const sourceBucket = new Bucket(this, 'Bucket');

const distribution = new CloudFrontWebDistribution(this, 'MyDistribution', {
  originConfigs: [
    {
      s3OriginSource: {
      s3BucketSource: sourceBucket
      },
      behaviors : [ {isDefaultBehavior: true}]
    }
  ],
  viewerCertificate: ViewerCertificate.fromAcmCertificate(myExampleCert, {
    aliases: ['mysubdomain.example.com']
  }
});
```

### Error Log

N/A, no error is given, but the distribution is created without using the recommended TLS protocol stated in the console.

### Environment

  - **CLI Version      :**  1.26.0 (build e251651)
  - **Framework Version:** 1.26.0
  - **OS               :** MacOS
  - **Language         :** English

### Other

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by blimmer**
(feat) Match console rec for ACM cert Security Policy (TLSv1.1_2016)
Closes #6547

cc @iliapolo

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by blimmer** [comment]
Hmm, based on the failure, it seems like I might need to do two PRs?

**Comment by blimmer** [comment]
I'm sorry for all the renames on this PR - it's unclear to me what the proper format is. Please let me know if this needs to be two PRs to retain both the feature and bugfix. I'm happy to rework in whatever way the process requires.

File: packages/@aws-cdk/aws-cloudfront/test/test.basic.ts [hunk 1]
```diff
@@ -572,6 +572,7 @@ export = {
             "Aliases": [],
```
Comment (blimmer) [inline packages/@aws-cdk/aws-cloudfront/test/test.basic.ts]: I'm also happy to break out different tests for defaults, as opposed to re-using the "basic usage" tests. I didn't quite know what the preference would be.

**Comment by iliapolo** [comment]
Hi @blimmer - Thanks again for this PR!

Like you already imagined, our process dictates that we do not combine bug fixes and features in the same PR. That just won't reflect properly in the CHANGELOG. Lets split these up.

**Comment by blimmer** [comment]
Thanks @iliapolo - The other commit will conflict with this one, so once this gets merged, I'll break out the bugfix into a new PR.

**Comment by blimmer** [comment]
Hmm something must have drifted or changed since I last worked on this PR. I'll have to investigate further later today or tomorrow.

**Comment by blimmer** [comment]
OK, I discovered the bug that I corrected between the two commits. This PR should be good to go now. Once it gets merged, I can break out the other PR to cherry-pick this bugfix commit: https://github.com/blimmer/aws-cdk/commit/314510facadc8c2a6fc7948b4890884cb529bfbb

**Comment by iliapolo** [comment]
@eladb can I get your input here? 

On one hand it makes sense to use the console recommended values. On the other hand, the current default is set by CloudFormation and uses the minimum possible version (which incidentally is also default *selection* in the UI). 

From the console:

> We recommend that you specify TLSv1.1_2016 unless your users are using browsers or devices that do not support TLSv1.1 or later.

I imagine this recommendation comes from security concerns, but as far as usability goes, using the least common denominator that supports most versions is better. 

@blimmer I'm starting to have second thoughts on this one due to the above reasoning. Thoughts?

**Comment by blimmer** [comment]
I can see the argument either way. In my opinion, I would prefer to default into the more secure recommendation but, as you mentioned, maybe that's what CloudFormation should be doing instead of CDK when the user isn't explicit.

If we do choose to go that route (closing this PR), we should also validate what the SNI default is. The code states that it defaults to SSL_V3, but I'm not sure.

https://github.com/aws/aws-cdk/blob/909b590b0c92c0f6807018d0441bd259cc696f70/packages/%40aws-cdk/aws-cloudfront/lib/web_distribution.ts#L70

IIRC SNI is really expensive, so I don't think I can check it myself.

**Comment by iliapolo** [comment]
@blimmer The code states its TLSv1 for SNI - which is true. I'm closing this PR for now. 
We are planning some substantial effort on this module in the upcoming future, so we will revisit this at that point.

**Comment by blimmer** [comment]
Sounds good - no worries. Thanks for considering.

**Pull request closed without merging**